tests/test_eks/test_eks.py

```diff
@@ -0,0 +1,770 @@
+        ACCOUNT_ID,
+        generated_test_data.cluster_name,
+        generated_test_data.nodegroup_names,
+        None,
+    ]
+
+    all_arn_values_should_be_valid(
+        expected_arn_values=expected_arn_values,
+        pattern=RegExTemplates.NODEGROUP_ARN,
+        arn_under_test=generated_test_data.nodegroup_describe_output[
+            NodegroupAttributes.ARN
+        ],
+    )
+
+
+@pytest.mark.freeze_time
+def test_create_nodegroup_generates_valid_nodegroup_created_timestamp(NodegroupBuilder):
+    _, generated_test_data = NodegroupBuilder()
+    current_time = iso_8601_datetime_without_milliseconds(datetime.now())
+
+    result_time = iso_8601_datetime_without_milliseconds(
+        generated_test_data.nodegroup_describe_output[NodegroupAttributes.CREATED_AT]
+    )
+
+    result_time.should.equal(current_time)
+
+
+@pytest.mark.freeze_time
+def test_create_nodegroup_generates_valid_nodegroup_modified_timestamp(
+    NodegroupBuilder,
+):
+    client, generated_test_data = NodegroupBuilder()
+    current_time = iso_8601_datetime_without_milliseconds(datetime.now())
+
+    result_time = iso_8601_datetime_without_milliseconds(
+        generated_test_data.nodegroup_describe_output[NodegroupAttributes.MODIFIED_AT]
+    )
+
+    result_time.should.equal(current_time)
+
+
+def test_create_nodegroup_generates_valid_autoscaling_group_name(NodegroupBuilder):
+    _, generated_test_data = NodegroupBuilder()
+    result_resources = generated_test_data.nodegroup_describe_output[
+        NodegroupAttributes.RESOURCES
+    ]
+
+    result_asg_name = result_resources[NodegroupAttributes.AUTOSCALING_GROUPS][0][
+        NodegroupAttributes.NAME
+    ]
+
+    RegExTemplates.NODEGROUP_ASG_NAME_PATTERN.match(result_asg_name).should.be.true
+
+
+def test_create_nodegroup_generates_valid_security_group_name(NodegroupBuilder):
+    _, generated_test_data = NodegroupBuilder()
+    result_resources = generated_test_data.nodegroup_describe_output[
+        NodegroupAttributes.RESOURCES
+    ]
+
+    result_security_group = result_resources[NodegroupAttributes.REMOTE_ACCESS_SG]
+
+    RegExTemplates.NODEGROUP_SECURITY_GROUP_NAME_PATTERN.match(
+        result_security_group
+    ).should.be.true
+
+
+def test_create_nodegroup_saves_provided_parameters(NodegroupBuilder):
+    _, generated_test_data = NodegroupBuilder(minimal=False)
+
+    for key, expected_value in generated_test_data.attributes_to_test:
+        generated_test_data.nodegroup_describe_output[key].should.equal(expected_value)
+
+
+def test_describe_nodegroup_throws_exception_when_cluster_not_found(NodegroupBuilder):
+    client, generated_test_data = NodegroupBuilder()
+    expected_exception = ResourceNotFoundException
+    expected_msg = CLUSTER_NOT_FOUND_MSG.format(
+        exception_name=expected_exception.TYPE,
+        method=MethodNames.DESCRIBE_NODEGROUP,
+        cluster_name=generated_test_data.nonexistent_cluster_name,
+    )
+
+    client.describe_nodegroup.when.called_with(
```

Comment:
Exceptions should be tested using the `with pytest.raises` paradigm, to verify the format is correct. 
Example: https://github.com/spulec/moto/blob/master/tests/test_sts/test_sts.py#L738

Reply:
Same for the other 11 `when.called_with` calls